File: ruby/src/person.rb
# This class contains personal data: id, name, date_of_birth, date_of_death

class Person

  attr_accessor :age

  def id= id
    @id = id
  end

  def name= name
    @name = name
  end

  def date_of_birth= date_of_birth
    @date_of_birth = date_of_birth
  end

  def date_of_death= date_of_death
    @date_of_death = date_of_death
  end

  def get_date_of_birth
    return @date_of_birth
  end

  def get_name
    return @name
  end

  def get_id
    return @id
  end

  def get_date_of_death
    return @date_of_death
  end
end

File: Java/src/MeetingMatcher/sk/fiit/stuba/ir/badal/model/Person.java
package sk.fiit.stuba.ir.badal.model;


import java.util.Date;

/**
 * @author Matej Bádal <matejbadal@gmail.com>
 */
public class Person {


	private String id;
	private String name;
	private Date dateOfBirth = null;
	private Date dateOfDeath = null;

	public String getName() {
		return this.name;
	}

	public Person setName(String name) {
		this.name = name;
		return this;
	}

	public Person setDateOfBirth(Date dateOfBirth) {
		this.dateOfBirth = dateOfBirth;
		return this;
	}

	public Date getDateOfBirth() {
		return this.dateOfBirth;
	}

	public Date getDateOfDeath() {
		return this.dateOfDeath;
	}

	public Person setDateOfDeath(Date dateOfDeath) {
		this.dateOfDeath = dateOfDeath;
		return this;
	}

	public String getId() {
		return id;
	}

	public Person setId(String id) {
		this.id = id;
		return this;
	}

	public String toString() {
		return "[" + this.name + ", " + this.dateOfBirth + ", " + this.dateOfDeath + "]";
	}

	public boolean isComplete() {
		if (this.id != null && this.name != null && this.dateOfBirth != null && this.dateOfDeath != null) {
			return true;
		}
		return false;
	}
}
